org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/ProjectConfigurationManager.java

```diff
@@ -488,8 +488,10 @@ private void updateProjectConfiguration(ProjectConfigurationRequest request, IPr
     MavenProject mavenProject = request.mavenProject();
     Properties mavenProperties = mavenProject.getProperties();
     String sourceEncoding = mavenProperties.getProperty("project.build.sourceEncoding");
-    log.debug("Setting encoding for project {}: {}", project.getName(), sourceEncoding); //$NON-NLS-1$
-    project.setDefaultCharset(sourceEncoding, monitor);
+    if(project.getDefaultCharset() == null || !project.getDefaultCharset().equals(sourceEncoding)) {
```

Comment:
```suggestion
    if(Objects.equals(project.getDefaultCharset(), sourceEncoding)) {
```
That should also cover the case that the project's encoding is already null and no encoding is specified and would resetting it to null again. Furthermore the project's charset is then only fetched once.

If you think that's fine it should be applied to all other changes as well.

Reply:
Done ;)